packages/mui-material/src/Alert/Alert.js

```diff
@@ -248,6 +254,23 @@ Alert.propTypes /* remove-proptypes */ = {
     PropTypes.oneOf(['error', 'info', 'success', 'warning']),
     PropTypes.string,
   ]),
+  /**
+   * The components used for each slot inside the Alert.
+   * Either a string to use a HTML element or a component.
+   * @default {}
+   */
+  components: PropTypes.shape({
+    CloseButton: PropTypes.elementType,
+    CloseIcon: PropTypes.elementType,
+  }),
+  /**
+   * The props used for each slot inside.
+   * @default {}
+   */
+  componentsProps: PropTypes.shape({
+    closeButton: PropTypes.object,
+    closeIcon: PropTypes.object,
```

Comment:
Should we use the same types as done in the `Backdrop` component? cc @michaldudak

Reply:
The prop types look the same to me here - I did slightly vary on the TS types because I'm actually not sure the intersection applied to `componentsProps.root` on Backdrop is actually what we want.

Maybe this is the actual question - wouldn't we want the AlertComponentsPropsOverrides object to intersect with the entire componentsProps object instead of the individual properties of said object?

For example, I want my close icon to be a div - shouldn't I be able to 

```ts
declare module '@mui/material/Alert' {
  interface AlertComponentsPropsOverrides { 
    closeIcon?: React.HTMLAttributes<HTMLDivElement>
  }
}
```

and expect the new type to be 

```ts
componentsProps: {
  closeButton?: IconButtonProps,
  closeIcon?: React.HTMLAttributes<HTMLDivElement> 
}
```
?

Reply:
@michaldudak can provide more information on what was the idea behind this type of override.

Reply:
I don't remember us even considering the override at the object level. It does make sense if you want to provide arbitrary components in `components` and control their props with `componentsProps`. I propose we make the implementation consistent with what we have currently and we can discuss the alternative implementation in another issue. Could you create one, @jake-collibra?

Reply:
So in this case would it be:

```ts
componentsProps: {
  closeButton?: IconButtonProps & AlertComponentPropsOverrides;
  closeIcon?: SvgIconProps & AlertComponentPropsOverrides; 
}
```
?

Or would we want to have different override interfaces for each prop? In this case I cannot think of a time when I'd want both to be the same type.

I can create an issue when I understand the suggested current approach better.